solidity/src/proof_plans/SortMergeJoinExec.pre.sol

```diff
@@ -0,0 +1,828 @@
+                    evaluate_input_plans(plan_ptr, builder_ptr, mload(add(left_right_and_output_evaluations, WORD_SIZE)))
+                i_eval :=
+                    consume_and_membership_check_right_column_evals(
+                        builder_ptr,
+                        alpha_and_beta,
+                        mload(join_evals),
+                        hat_evals,
+                        res_chi_eval,
+                        chi_eval,
+                        left_right_and_output_evaluations
+                    )
+                mstore(add(left_right_and_output_evaluations, WORD_SIZE), join_evals)
+
+                monotonic_verify(
+                    builder_ptr,
+                    mload(alpha_and_beta),
+                    mload(add(alpha_and_beta, WORD_SIZE)),
+                    i_eval,
+                    res_chi_eval,
+                    1,
+                    1
+                )
+                plan_ptr_out := plan_ptr
+            }
+
+            function evaluate_remaining_check_on_both_left_and_right_sides(
+                plan_ptr, builder_ptr, alpha_and_beta, left_right_and_output_evaluations, left_chi_eval, right_chi_eval
+            ) -> plan_ptr_out, w_eval {
+                let u_column_eval_array, u_chi_eval :=
+                    evaluate_u_column_with_monotony_check(builder_ptr, alpha_and_beta)
+                w_eval :=
+                    membership_check_evaluate(
+                        builder_ptr,
+                        mload(alpha_and_beta),
+                        mload(add(alpha_and_beta, WORD_SIZE)),
+                        u_chi_eval,
+                        left_chi_eval,
+                        u_column_eval_array,
+                        mload(left_right_and_output_evaluations)
+                    )
+                w_eval :=
+                    mulmod_bn254(
+                        w_eval,
+                        membership_check_evaluate(
+                            builder_ptr,
+                            mload(alpha_and_beta),
+                            mload(add(alpha_and_beta, WORD_SIZE)),
+                            u_chi_eval,
+                            right_chi_eval,
+                            u_column_eval_array,
+                            mload(add(left_right_and_output_evaluations, WORD_SIZE))
+                        )
+                    )
+                plan_ptr_out := plan_ptr
+            }
+
+            function evaluate_sort_merge_join_outputs(
+                plan_ptr, builder_ptr, left_right_and_output_evaluations, left_chi_eval, right_chi_eval, output_chi_eval
+            ) -> plan_ptr_out {
+                let alpha_and_beta := mload(FREE_PTR)
+                mstore(FREE_PTR, add(alpha_and_beta, WORDX2_SIZE))
+                mstore(alpha_and_beta, builder_consume_challenge(builder_ptr))
+                mstore(add(alpha_and_beta, WORD_SIZE), builder_consume_challenge(builder_ptr))
```

Comment:
Please add some comment. Yes I understand why you did this (saving one local variable slot) but people will forget over time.